Tests/SwiftSyntaxBuilderTest/StringLiteralTests.swift

```diff
@@ -170,4 +170,182 @@ final class StringLiteralTests: XCTestCase {
       """#
     )
   }
+
+  func testStringLiteralInExpr() {
+    let buildable = ExprSyntax(
+      #"""
+      "Validation failures: \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"
+      """#
+    )
+
+    AssertBuildResult(
+      buildable,
+      #"""
+      "Validation failures: \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"
+      """#
+    )
+  }
+
+  func testStringSegmentWithCode() {
+    let buildable = StringSegmentSyntax(content: .stringSegment(#"\(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"#))
+
+    AssertBuildResult(
+      buildable,
+      #"\(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"#
+    )
+  }
+
+  func testStringLiteralSegmentWithCode() {
+    let buildable = StringLiteralSegmentsSyntax {
+      StringSegmentSyntax(content: .stringSegment(#"Error validating child at index \(index) of \(nodeKind):"#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"Node did not satisfy any node choice requirement."#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"Validation failures:"#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"\(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"#))
```

Comment:
If you want to form the syntax tree as it’s being parsed, this should be a `ExpressionSegmentSyntax` instead of a `StringSegmentSyntax`. Similarly, the interpolations in the first line should be `ExpressionSegmentSyntax` nodes. I’d be fine to just use tests that invoke the parser, like `testStringLiteralInExpr` if that’s simpler.